ctapipe_io_lst/__init__.py

```diff
@@ -256,44 +278,54 @@ def create_subarray(self, tel_id=1):
         # LSTs telescope position taken from MC from the moment
         tel_positions = {tel_id: [50., 50., 16] * u.m}
 
-        subarray = SubarrayDescription("LST1 subarray")
-        subarray.tel_descriptions = tel_descriptions
-        subarray.positions = tel_positions
-        subarray.tel[tel_id] = lst_tel_descr
+        subarray = SubarrayDescription(
+            name=f"LST-{tel_id} subarray",
+            tel_descriptions=tel_descriptions,
+            tel_positions=tel_positions,
+        )
 
         return subarray
 
     def _generator(self):
 
         # container for LST data
-        self.data = LSTDataContainer()
-        self.data.meta['input_url'] = self.input_url
-        self.data.meta['max_events'] = self.max_events
-        self.data.meta['origin'] = 'LSTCAM'
+        array_event = LSTArrayEventContainer()
+        array_event.meta['input_url'] = self.input_url
+        array_event.meta['max_events'] = self.max_events
+        array_event.meta['origin'] = 'LSTCAM'
 
         # fill LST data from the CameraConfig table
-        self.fill_lst_service_container_from_zfile()
+        array_event.lst.tel[self.tel_id].svc = self.fill_lst_service_container()
 
         # initialize general monitoring container
-        self.initialize_mon_container()
+        self.initialize_mon_container(array_event)
 
         # loop on events
-        for count, event in enumerate(self.multi_file):
+        for count, zfits_event in enumerate(self.multi_file):
+            array_event.count = count
+            array_event.index.event_id = zfits_event.event_id
```

Comment:
Since event_id == 0 is practically always corrupted I would find here a way to skip it

Reply:
`event_id == 0` does not exist. The first event in the file has `event_id == 1`.

Are you saying you want to skip the first event if it has `event_id == 0` or to always skip the first event (`event_id == 1`)?

I would not skip events in the source. Let us deal with it at a later stage.

Reply:
I meant the first event, indeed one can not skip here, it would be a problem for the dr4s correction. Ok, let deal with it in later stage